source/Cosmos.HAL2/Drivers/PCI/Audio/AudioManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Cosmos.HAL.Drivers;
+using Cosmos.HAL.Drivers.PCI.Audio.Generic;
+using System;
+using System.Collections.Generic;
+
+namespace Cosmos.HAL.Drivers.PCI.Audio
+{
+    /// <summary>
+    /// Audio mananger class. Right now, it only supports sound blaster 16.
+    /// </summary>
+    public class AudioManager
+    {
+        public static List<SoundCard> devices = new List<SoundCard>();
+
+        public static void Init()
+        {
+            devices.Clear(); //just in case
+            try
+            {
+                Console.WriteLine("Searching for sound cards");
+
+                var sb16 = new SoundBlaster16();
+                sb16.Enable();
+                if (sb16.IsPresent)
+                {
+                    Console.WriteLine("Found Sound blaster 16!");
+                    devices.Add(sb16);
+                }
+
+                if (devices.Count == 0)
+                {
+                    Console.WriteLine("No supported sound cards sound.");
+                }
+                else
+                {
+                    Console.WriteLine("Sound card(s) initialized successfully!");
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error while initializing sound: " + ex.Message);
```

Comment:
Does the sound blaster init regularly throw exceptions? If not please dont use a try catch block, since cosmos currently cant handle multilayered try catch blocks.